### ISSUE TITLE
Function to rename an existing file with a zd file name

### ISSUE THREAD

**Issue by anthrolisp** [EFLS/zetteldeft#13]
Function to rename an existing file (that's without a zd file naming structure) with a zd file name.

**Comment by EFLS** [comment]
That would only make sense if the file itself is already located within the `zetteldeft` directory. Or is your suggestion to move or copy the file as well?

I think this falls outside the scope of `zetteldeft`, but I'm open to hear why you think otherwise.

**Comment by anthrolisp** [comment]
Correct, the original context was for moving previous existing files/zettels into zetteldeft.

**Comment by EFLS** [comment]
You could always write some additional helper function yourself.

Generate an ID with `zd-generate-id` and use that to rename the file, maybe prepending it to the current filename.

**Comment by gklimowicz** [comment]
I've just gone through a couple of trials migrating existing files into `zetteldeft` directory (well, multiple; I have a couple of very different domains, each as its own Zettelkasten for the moment).

My elisp skills are not great, and since this was largely a one-time thing, I just wrote some shell scripts to handle it (even mocking up IDs for files based on either dates and times within the file or the file's creation).

The scripts make use of another shell script I have that can be used to rename files and file titles throughout the collection of files (including relinking or replacing references to titles). I did this in the shell partly to be able to easily accommodate git file renames.
